infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftInternalEnvironment.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.environment;
+
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.core.model.workspace.config.Environment;
+import org.eclipse.che.api.installer.server.InstallerRegistry;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.InternalEnvironment;
+import org.eclipse.che.api.workspace.server.spi.RecipeRetriever;
+
+/** @author Sergii Leshchenko */
+public class OpenShiftInternalEnvironment extends InternalEnvironment {
+
+  private final OpenShiftEnvironmentParser openShiftEnvironmentParser;
+  private OpenShiftEnvironment osEnv;
```

Comment:
I don't see any reason to have both `OpenShiftInternalEnvironment` and `OpenShiftEnvironment`. I think `OpenShiftInternalEnvironment` should be POJO object and I propose the following changes
```
public class OpenShiftInternalEnvironment extends InternalEnvironment {
  private final Map<String, Pod> pods;
  private final Map<String, Service> services;
  private final Map<String, Route> routes;
  private final Map<String, PersistentVolumeClaim> persistentVolumeClaims;

  // getters are here
}
```
Then OpenShiftInternalEnvironmentFactory can parse it with OpenShiftEnvironmentParser.
WDYT?

Reply:
Agree, moreover, I think OpenShiftEnvironmentParser's functionality can be a part of OpenShiftInternalEnvironmentFactory?